catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -0,0 +1,231 @@
+     *
+     * @param upstreamResponse - a SourceResponse with a list of {@link Metacard}s to compress
+     * @param arguments        - a map of arguments to use for processing.  This method expects "fileName" to be set
+     * @return - a {@link BinaryContent} item with the {@link InputStream} for the Zip file
+     * @throws CatalogTransformerException when the transformation fails
+     */
+    public BinaryContent transform(SourceResponse upstreamResponse,
+            Map<String, Serializable> arguments) throws CatalogTransformerException {
+
+        if (upstreamResponse == null || upstreamResponse.getResults() == null ||
+                upstreamResponse.getResults()
+                        .size() == 0) {
+            throw new CatalogTransformerException("No Metacards were found to transform.");
+        }
+
+        if (MapUtils.isEmpty(arguments) || !arguments.containsKey(ZipDecompression.FILE_PATH)) {
+            throw new CatalogTransformerException("No 'filePath' argument found in arguments map.");
+        }
+
+        ZipFile zipFile;
+        String filePath = (String) arguments.get(ZipDecompression.FILE_PATH);
+        try {
+            zipFile = new ZipFile(filePath);
+        } catch (ZipException e) {
+            LOGGER.debug("Unable to create zip file with path : {}", filePath, e);
+            throw new CatalogTransformerException(String.format("Unable to create zip file at %s",
+                    filePath), e);
+        }
+
+        List<Result> resultList = upstreamResponse.getResults();
+
+        Map<Resource, String> resourceList = new HashMap<>();
+
+        for (Result result : resultList) {
+            Metacard metacard = result.getMetacard();
+
+            try {
+                ZipParameters zipParameters = new ZipParameters();
+                zipParameters.setSourceExternalStream(true);
+                zipParameters.setFileNameInZip(metacard.getId());
+
+                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                ObjectOutputStream objectOutputStream =
+                        new ObjectOutputStream(byteArrayOutputStream);
+                objectOutputStream.writeObject(new MetacardImpl(metacard));
+                objectOutputStream.flush();
+                objectOutputStream.close();
+
+                InputStream inputStream =
+                        new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+
+                zipFile.addStream(inputStream, zipParameters);
+
+                if (hasLocalResources(metacard)) {
+                    resourceList.putAll(getAllMetacardContent(metacard));
+                }
+
+            } catch (IOException | ZipException e) {
+                LOGGER.debug("Failed to add metacard with id {}.}", metacard.getId(), e);
+            }
+        }
+
+        for (Map.Entry<Resource, String> entry : resourceList.entrySet()) {
```

Comment:
Could use `forEach` on the map to make this a bit simpler. Also it's a bit confusing that `resourceList` is actually a `Map`.